packages/components/src/mobile/bottom-sheet/test/cell.native.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+import { fireEvent, render } from 'test/helpers';
+
+/**
+ * Internal dependencies
+ */
+import BottomSheetCell from '../cell.native.js';
+
+describe( '<BottomSheetCell />', () => {
+	it( 'renders the component', async () => {
+		const screen = render( <BottomSheetCell /> );
+		const cellTouchable = screen.getByTestId( 'cell-touchable' );
+
+		expect( cellTouchable ).toBeTruthy();
+	} );
+
+	it( 'displays a clear button when the text has value', async () => {
+		const onChangeValueMock = jest.fn();
+		const testValueString = 'test value string';
+		const newTestValueString = 'new test value string';
+
+		const screen = render(
+			<BottomSheetCell
+				displayClearButton={ true }
+				value={ testValueString }
+				onChangeValue={ onChangeValueMock }
+			/>
+		);
+
+		// Invoke editing behavior by pressing the Cell
+		const cellTouchable = await screen.getByTestId( 'cell-touchable' );
+		fireEvent.press( cellTouchable );
+
+		// Simulate onChangeValue events (without rerender)
+		const textInput = await screen.getByTestId( 'text-input' );
+		fireEvent( textInput, 'focus' );
+		fireEvent.changeText( textInput, newTestValueString );
+		expect( onChangeValueMock ).toHaveBeenCalledWith( newTestValueString );
+
+		const clearButton = await screen.getByTestId( 'clear-button' );
+		expect( clearButton ).toBeTruthy();
+	} );
+
+	it( 'does NOT display a clear button when the text is empty', async () => {
+		const onChangeValueMock = jest.fn();
+
+		const screen = render(
+			<BottomSheetCell
+				displayClearButton={ true }
+				value={ '' }
+				onChangeValue={ onChangeValueMock }
+			/>
+		);
+
+		// Invoke editing behavior by pressing the Cell
+		const cellTouchable = await screen.getByTestId( 'cell-touchable' );
+		fireEvent.press( cellTouchable );
+
+		const clearButton = await screen.queryByTestId( 'clear-button' );
+		expect( clearButton ).toBeNull();
+	} );
+
+	it( 'does NOT display a clear button when displayClearButton is false', async () => {
+		const onChangeValueMock = jest.fn();
+
+		const screen = render(
+			<BottomSheetCell
+				displayClearButton={ false }
+				value={ '' }
+				onChangeValue={ onChangeValueMock }
+			/>
+		);
+
+		const clearButton = await screen.queryByTestId( 'clear-button' );
+		expect( clearButton ).toBeNull();
+	});
+
+	it( 'clears text input value when the clear button is pressed', async () => {
```

Comment:
This test is checking if the handler is called, which doesn't guarantee that the text input would be actually cleared. I'm wondering if we should update the test's description in order to be more accurate regarding what it's covering, wdyt?

Reply:
Good call! I've updated the test description here to reflect that only the `onClear` handler will be called.

Reply:
Great, thanks for updating it 🙇 .